### ISSUE TITLE
hlint confused by qualified ++ type operators

### ISSUE THREAD

**Issue by bpfoley** [ndmitchell/hlint#400]
HEAD has a couple of problems parsing ++ as a type operator when it's qualified. This possibly applies to all qualified type operators with symbolic names.

The issue can be seen in the example below

```haskell
{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE TypeOperators #-}                                                  

module N where

type (++) a b = (a,b)

-- Correctly parses OK in GHCi & hlint
f1 :: Eq (Bool ++ Bool) => a
f1 = undefined

-- Correctly parses OK in GHCi & hlint
f2 :: Eq ((N.++) Bool Bool) => a
f2 = undefined

-- Parses OK in GHCi, but incorrectly a parse error in hlint
f3 :: Eq (Bool N.++ Bool) => a
f3 = undefined

-- Parse error in GHCi (because it's a prefix op), but incorrectly
-- parses OK in hlint
f4 :: Eq (Bool (N.++) Bool) => a
f4 = undefined
```

**Comment by ndmitchell** [comment]
Thanks for the report - this is an upstream bug in haskell-src-exts, and I've moved your good set of test cases to their tracker: https://github.com/haskell-suite/haskell-src-exts/issues/79.

**Comment by templateK** [comment]
@ndmitchell  By looking at https://github.com/haskell-suite/haskell-src-exts/issues/79, 
I don't expect upstream fix anytime soon. 
So, Is there any workaround which can be done on hlint side?

**Comment by ndmitchell** [comment]
@templateK yes, details in about a month.

**Comment by ndmitchell** [comment]
Confirmed works with HEAD that doesn't use HSE for parsing, so fixed in the next release.
